selfdrive/car/tesla/ACC_module.py

```diff
@@ -258,7 +258,10 @@ def _calc_follow_button(self, CS, lead_car,speed_limit_kph, speed_limit_valid, s
     # Relative velocity between the lead car and our set cruise speed.
     future_vrel_kph = lead_speed_kph - CS.v_cruise_actual
     # How much we can accelerate without exceeding the max allowed speed.
-    max_acc_speed_kph = self.max_v_by_speed_limit(CS.v_cruise_actual * CV.KPH_TO_MS, self.acc_speed_kph, CS) * CV.MS_TO_KPH
+    if (CS.maxdrivespeed > 0)  and CS.useTeslaMapData and (CS.mapAwareSpeed or (CS.baseMapSpeedLimitMPS <2.7)):

```

Comment:
Does this logic need to be removed from `max_v_by_speed_limit` now that you extracted it, or would it make sense to push these changes into it?

Reply:
Currently needs to remain, but I will look at some ways to clean up the code.

Reply:
why are we switching the max speed to actual? CS.v_cruise_actual is what's set in DI for CC. the max_v_by_speed_limit needs to take a maximum speed limit as the second parameter. This code would send the DI set cruise (which changes as we adjust) as max. It should send the acc_speed_kph which is the max we set. I do not see how this would work. So will need more info.